src/commands/producers.ts

```diff
@@ -66,16 +72,21 @@ export class ProduceRecordCommandHandler {
                     const randomizedKey = (key) ? faker.fake(key) : key;
                     faker.seed(seed);
                     const randomizedValue = faker.fake(value);
+                    if (headers && headers.size > 0) {
```

Comment:
if you want to use the same seed for headers, as you documented, then you need to:
```json
Object.keys(messageHeaders).forEach(val => {
   faker.seed(seed);
   messageHeaders[val] = faker.fake(messageHeaders[val] as string;
});

```